test/e2e/declarative_test_fwk/helper.py

```diff
@@ -201,16 +235,50 @@ def assert_items(self, expectations: dict, state: dict) -> None:
         if not expectations:
             # nothing to assert as there are no expectations
             return
-        if not state:
-            # there are expectations but no given state to validate
+        if not actual:
+            # there are expectations but no given actual state to validate
             # following assert will fail and assert introspection will provide useful information for debugging
-            assert expectations == state
+            assert expectations == actual
 
         for (key, value) in expectations.items():
             # conditions are processed separately
             if key == "conditions":
                 continue
-            assert (key, value) == (key, state.get(key))
+            # tags are processed by custom logic
+            if key == "tags":
+                for expectedTag in value:
+                    tagKeyExist = False
+                    tagKey = expectedTag.get("key")
+                    latestTags = actual.get(key)
+                    for latestTag in latestTags:
+                        if ("key", tagKey) in latestTag.items():
+                            assert expectedTag == latestTag
+                            tagKeyExist = True
+                            break
+                    if tagKeyExist:
+                        continue
+                    assert expectedTag == None
+                continue
+            # parameterNameValues are processed by custom logic
+            if key == "parameterNameValues":
```

Comment:
Should this method be in the `ResourceHelper` class? This condition is only valid for the `ParameterGroup` implementation?

Reply:
Currently this condition is only valid for `ParameterGroup`. Other resources don't have this field, so they will skip this condition. However, describe_parameters is a independent API which is not related to describe_<resource type> APIs, so I put it into `ResourceHelper` class.

Reply:
If this condition is only valid for `ParameterGroup`, then it should live in the helper for that class. If I was trying to debug the code for the `ParameterGroup` tests, I wouldn't think to look here for that logic.

Reply:
Added `assert_extra_items_k8s` function to handle it. Override the function in `ResourceHelper` for `ParameterGroup`